graphql/resolvers.go

```diff
@@ -1064,6 +1067,49 @@ func (r *mutationResolver) SaveSubscription(ctx context.Context, subscription re
 	return true, nil
 }
 
+func (r *mutationResolver) UpdateUserSettings(ctx context.Context, userSettings *restModel.APIUserSettings) (bool, error) {
+	usr := route.MustHaveUser(ctx)
+	adminSettings, err := evergreen.GetConfig()
+	if err != nil {
+		return false, errors.Wrap(err, "Error retrieving Evergreen settings")
+	}
+	changedSettings, err := restModel.ApplyUserChanges(usr.Settings, *userSettings)
+	if err != nil {
+		return false, InternalServerError.Send(ctx, fmt.Sprintf("problem applying user settings: %s", err))
+	}
+	userSettingsInterface, err := changedSettings.ToService()
+	if err != nil {
+		return false, InternalServerError.Send(ctx, fmt.Sprintf("Error parsing user settings: %s", err))
+	}
+	UserSettings, ok := userSettingsInterface.(user.UserSettings)
+	if !ok {
+		return false, InternalServerError.Send(ctx, fmt.Sprintf("Unable to parse settings object: %s", err))
+	}
+
+	if len(UserSettings.GithubUser.LastKnownAs) == 0 {
```

Comment:
This block doesn't do anything. I don't think it's needed

Reply:
This is for the case that the user passes in an empty string as there github user name so we want to clear their `UserSettings.GithubUser`.

Reply:
And avoid running the logic to update their github user on an empty string.